src/libs/ReportUtils.ts

```diff
@@ -216,278 +237,208 @@ function isCanceledTaskReport(report = {}, parentReportAction = {}) {
 /**
  * Checks if a report is an open task report.
  *
- * @param {Object} report
- * @param {Object} parentReportAction - The parent report action of the report (Used to check if the task has been canceled)
- * @returns {Boolean}
+ * @param report
+ * @param parentReportAction - The parent report action of the report (Used to check if the task has been canceled)
  */
-function isOpenTaskReport(report, parentReportAction = {}) {
-    return isTaskReport(report) && !isCanceledTaskReport(report, parentReportAction) && report.stateNum === CONST.REPORT.STATE_NUM.OPEN && report.statusNum === CONST.REPORT.STATUS.OPEN;
+function isOpenTaskReport(report: OnyxEntry<Report>, parentReportAction?: OnyxEntry<ReportAction>): boolean {
+    return isTaskReport(report) && !isCanceledTaskReport(report, parentReportAction) && report?.stateNum === CONST.REPORT.STATE_NUM.OPEN && report?.statusNum === CONST.REPORT.STATUS.OPEN;
 }
 
 /**
  * Checks if a report is a completed task report.
- *
- * @param {Object} report
- * @returns {Boolean}
  */
-function isCompletedTaskReport(report) {
-    return isTaskReport(report) && report.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED && report.statusNum === CONST.REPORT.STATUS.APPROVED;
+function isCompletedTaskReport(report: OnyxEntry<Report>) {
+    return isTaskReport(report) && report?.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED && report?.statusNum === CONST.REPORT.STATUS.APPROVED;
 }
 
 /**
  * Checks if the current user is the manager of the supplied report
- *
- * @param {Object} report
- * @returns {Boolean}
  */
-function isReportManager(report) {
-    return report && report.managerID === currentUserAccountID;
+function isReportManager(report: OnyxEntry<Report>): boolean {
+    return report?.managerID === currentUserAccountID;
 }
 
 /**
  * Checks if the supplied report has been approved
- *
- * @param {Object} report
- * @returns {Boolean}
  */
-function isReportApproved(report) {
-    return report && report.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED && report.statusNum === CONST.REPORT.STATUS.APPROVED;
+function isReportApproved(report: OnyxEntry<Report> | undefined): boolean {
```

Comment:
I see in other cases you don't consider `report` as `undefined`.

Reply:
Actually we should consider report as undefined in some places , in several function we are getting report from `getReport` function which could report `undefined` 
<img width="952" alt="image" src="https://github.com/Expensify/App/assets/25230417/fc4c6742-1ebe-44ab-809c-51a2ef3b31b0">